lexicon/providers/godaddy.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import requests
+import json
+
+from .base import Provider as BaseProvider
+
+LOGGER = logging.getLogger(__name__)
+
+def ProviderParser(subparser):
+    subparser.add_argument('--auth-key', help='specify the key to access the API')
+    subparser.add_argument('--auth-secret', help='specify the secret to access the API')
+
+class Provider(BaseProvider):
+
+    def __init__(self, options, engine_overrides=None):
+        super(Provider, self).__init__(options, engine_overrides)
+        self.domain_id = None
+        self.api_endpoint = self.engine_overrides.get('api_endpoint', 'https://api.godaddy.com/v1')
+
+    def authenticate(self):
+        domain = self.options.get('domain')
+
+        result = self._get('/domains/{0}'.format(domain))
+        self.domain_id = result['domainId']
+
+    def create_record(self, type, name, content):
+        domain = self.options.get('domain')
+        ttl = self.options.get('ttl')
+
+        data = {'data': content}
+        if ttl:
+            data['ttl'] = ttl
+
+        self._put('/domains/{0}/records/{1}/{2}'
+                  .format(domain, type, self._relative_name(name)), [data])
+
+        LOGGER.debug('create_record: %s %s %s', type, name, content)
+
+        return True
+
+    def list_records(self, type=None, name=None, content=None):
+        domain = self.options.get('domain')
+        records = []
+
+        url = '/domains/{0}/records'.format(domain)
+        if type:
+            url += '/{0}'.format(type)
+        if name:
+            url += '/{0}'.format(self._relative_name(name))
+
+        raws = self._get(url)
+
+        for raw in raws:
+            records.append({
+                'type': raw['type'],
+                'name': self._full_name(raw['name']),
+                'ttl': raw['ttl'],
+                'content': raw['data']
+            })
+
+        if content:
+            records = [record for record in records if record['data'].lower() == content.lower()]
+
+        LOGGER.debug('list_records: %s', records)
+
+        return records
+
+    def update_record(self, identifier, type=None, name=None, content=None):
+        # With GoDaddy API, creating a record for given type and name is the same
+        #   than updating the record.
+        return self.create_record(type, name, content)
+
+    def delete_record(self, identifier=None, type=None, name=None, content=None):
+        domain = self.options.get('domain')
+
+        if not type:
+            raise Exception('ERROR: type must be setted')
```

Comment:
these exceptions should probably be:

`raise Exception('Error: type is required')` just to be consistent. 

I'll probably have to think more about a consistent Exception pattern and input validation.